src/http/request.cr

```diff
@@ -68,4 +89,21 @@ class HTTP::Request
   private def uri
     (@uri ||= URI.parse(@resource)).not_nil!
   end
+
+  private def ips_from(header)
+    if ips = headers[header]? || headers["Http-#{header.tr("_", "-")}"]?
+      ips.strip.split(/[,\s]+/)
+    else
+      [] of String
+    end
+  end
+
+  private def first_remote_ip_address(ip_addresses)
+    ip_addresses.find { |ip| !trusted_proxy?(ip) }
+  end
+
+  private def trusted_proxy?(ip)
+    ip =~ /\A127\.0\.0\.1\Z|\A(10|172\.(1[6-9]|2[0-9]|30|31)|192\.168)\.|\A::1\Z|\Afd[0-9a-f]{2}:.+|\Alocalhost\Z|\Aunix\Z|\Aunix:/i
```

Comment:
*  `127.0.0.0/8` https://tools.ietf.org/html/rfc6890
* Should we trust link-local addresses? `fe80::/10`

I also wonder if given that we shouldn't have `remote_ip(trust_headers=true, trust_all_clients=false)`?